rcl/include/rcl/type_version_hash.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__TYPE_VERSION_HASH_H_
+#define RCL__TYPE_VERSION_HASH_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+#include "rcutils/sha256.h"
+#include "type_description_interfaces/msg/type_description.h"
+
+/// Convert
```

Comment:
Can we make this more descriptive?  Something like:

```suggestion
/// Given a TypeDescription message, output a string of the hashable JSON data.
```